src/EnergyPlus/WaterCoils.cc

```diff
@@ -6377,6 +6382,185 @@ Label10: ;
+		int & CoilNum // index to heating coil
+	)
+	{
+		// Using/Aliasing
+		using DataPlant::ScanPlantLoopsForObject;
+		using General::RoundSigDigits;
+		using General::SolveRegulaFalsi;
+		using PlantUtilities::SetComponentFlowRate;
+
+		// SUBROUTINE PARAMETER DEFINITIONS:
+		static std::string const RoutineName( "CalcWaterCoilWaterFlowRate: " ); // for error messages
+		int const MaxIter( 200 ); // Maximum number of iterations for solver
+		Real64 const Acc( 1.0e-3 ); // Accuracy of solver result
+									
+		// SUBROUTINE LOCAL VARIABLE DECLARATIONS:
+		Array1D< Real64 > Par( 6 ); // Parameter array passed to solver
+		int AirOutletNode; // node number of coil outlet air node
+		int WaterInletNode; // node number of coil inlet water node
+		int WaterOutletNode; // node number of coil outlet water node
+		Real64 TempSetPoint; // target coil outlet node temperature set point [C]
+		Real64 mdot; // local water mass flow rate [kg/s]
+		Real64 mdotLow; // saved low water mass flow rate [kg/s]
+		bool errFlag; // warning flag
+		Real64 PartLoadFrac; // local fraction of max water flow rate
+		Real64 minPartLoadFrac; // minPLR pass to solver, based on max water flow rate
+		Real64 maxPartLoadFrac; // manPLR pass to solver, based on max water flow rate
+		int SolFlag; // solver return flag, -1 = max iterations exceeded, -2 = solution not found, other = number of iterations
+		Real64 deltaT; // air temperature difference from temperature set point [C]
+		std::string WaterCoilType; // used for warning messages
+
+		if ( WaterCoil( CoilNum ).WaterLoopCompNum == 0 ) {
+			ScanPlantLoopsForObject( WaterCoil( CoilNum ).Name, WaterCoil( CoilNum ).WaterCoilType_Num, WaterCoil( CoilNum ).WaterLoopNum, WaterCoil( CoilNum ).WaterLoopSide, WaterCoil( CoilNum ).WaterLoopBranchNum, WaterCoil( CoilNum ).WaterLoopCompNum, _, _, _, _, _, errFlag );
+			if ( WaterCoil( CoilNum ).WaterLoopCompNum == 0 ) return;
+		}
+
+		AirOutletNode = WaterCoil( CoilNum ).AirOutletNodeNum;
+		WaterInletNode = WaterCoil( CoilNum ).WaterInletNodeNum;
+		WaterOutletNode = WaterCoil( CoilNum ).WaterOutletNodeNum;
+		TempSetPoint = Node( AirOutletNode ).TempSetPoint;
+
+		mdot = 0.0;
+		SetComponentFlowRate( mdot, WaterInletNode, WaterOutletNode, WaterCoil( CoilNum ).WaterLoopNum, WaterCoil( CoilNum ).WaterLoopSide, WaterCoil( CoilNum ).WaterLoopBranchNum, WaterCoil( CoilNum ).WaterLoopCompNum );
+		mdotLow = mdot;
+		SimulateWaterCoilComponents( CoilName, FirstHVACIteration, CoilNum );
+
+		if ( WaterCoil( CoilNum ).WaterCoilType_Num == WaterCoil_SimpleHeating ) { // heating coil
+			deltaT = TempSetPoint - Node( AirOutletNode ).Temp; // positive result means coil is on
+		} else {
+			deltaT = Node( AirOutletNode ).Temp - TempSetPoint; // positive result means coil is on
+		}
+
+// *** NEED TO FIGURE OUT LOCKOUT LOGIC for non-OA system coils ( and OA system coils if they can be locked out ) *** and bypass next calculations if in lockout mode //
```

Comment:
@mjwitte when should OA System water coils, or any coil in the OA System for that matter, be locked out?